### PR TITLE
Emit `used-before-assignment` for further imports guarded by TYPE_CHECKING

### LINKED ISSUE #7539: Audit remaining uses of `TYPING_TYPE_CHECKS_GUARDS`
Along the lines of #7524 and #6787, there is probably a false positive for `used-before-assignment`/`unused-variable` lurking in the use of `defstmt_parent.test.as_string() in TYPING_TYPE_CHECKS_GUARDS`, which is not robust against importing `typing` under an alias.

I advised us against doing a drop-in replacement by `in_type_checking_block` without writing more tests. For discussion on that see https://github.com/PyCQA/pylint/pull/7525#discussion_r982354594.

To close this issue, we need a contributor to write a regression test that fails on `main` and ideally a test that also would have failed if we had taken the simple approach of just doing the drop-in replacement in https://github.com/PyCQA/pylint/pull/7525#discussion_r982024065.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Previously, this message was only emitted for imports guarded directly under TYPE_CHECKING, not guarded two if-branches deep, nor when TYPE_CHECKING was imported from typing under an alias.

Closes #7539

### CODE REVIEW

**Review by Pierre-Sassoulas — DISMISSED** [review]
``in_type_checking_block`` is indeed a lot more accurate 😄

File: tests/functional/u/used/used_before_assignment_typing.py [hunk 1]
```diff
@@ -74,3 +78,12 @@ def function(self, var: int) -> None:
 
     def other_function(self) -> None:
         _x: MyThirdClass = self
+
+
+class MyFourthClass:  # pylint: disable=too-few-public-methods
+    """Class to test conditional imports guarded by TYPE_CHECKING two levels
+    up then used in function annotation. See https://github.com/PyCQA/pylint/issues/7539"""
+
+    def comparator(self, comparator: math.isclose, first, second):  # [used-before-assignment]
```
Comment (jacobtylerwalls) [inline tests/functional/u/used/used_before_assignment_typing.py:87]:
```suggestion
    def is_close(self, comparator: math.isclose, first, second):  # [used-before-assignment]
```

**Comment by jacobtylerwalls** [comment]
The false positive in flask is already documented at #7609. It's true that this PR will ever so slightly drive more traffic into that false positive, but I don't think that's a reason not to merge it.

**Comment by jacobtylerwalls** [comment]
@Pierre-Sassoulas This one's ready for review, if you get a chance.

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM 👌 Thanks for the reminder !

**Pull request merged**